packages/browser/src/tracing/browserTracingIntegration.ts

```diff
@@ -356,6 +385,22 @@ export const browserTracingIntegration = ((_options: Partial<BrowserTracingOptio
         });
       });
 
+      if (enablePreviousTrace) {
+        let previousTraceInfo = persistPreviousTrace ? getPreviousTraceFromSessionStorage() : undefined;
+
+        client.on('spanStart', span => {
+          if (getRootSpan(span) !== span) {
+            return;
+          }
+
+          previousTraceInfo = addPreviousTraceSpanLink(previousTraceInfo, span);
+
+          if (persistPreviousTrace) {
+            storePreviousTraceInSessionStorage(previousTraceInfo);
+          }
```

Comment:
I think all this code around the previous trace can be put in here. Then there is also no need to mutate the variable or store `undefined` in a variable if persisting is turned off.

```suggestion

          if (persistPreviousTrace) {
            const updatedPreviousTraceInfo = addPreviousTraceSpanLink(
                getPreviousTraceFromSessionStorage(), span
             )
                  
            storePreviousTraceInSessionStorage(updatedPreviousTraceInfo);
          }
```

Reply:
Happy to rewrite this but I still need to call `addPreviousTraceSpanLink` if `persistPreviousTrace` is false (as by default). Let me take another look

Reply:
I rewrote this a bit based on your suggestion as well as on unifying the two options. [b44e790](https://github.com/getsentry/sentry-javascript/pull/15569/commits/b44e790babb1f8d4365f667047d6afa309ebb044)